clash-ghc/src-ghc/Clash/GHC/LoadModules.hs

```diff
@@ -702,3 +706,27 @@ removeStrictnessAnnotations pm =
 
     -- rmConDeclF :: GHC.DataId name => GHC.ConDeclField name -> GHC.ConDeclField name
     rmConDeclF cdf = cdf {GHC.cd_fld_type = rmHsType (GHC.cd_fld_type cdf)}
+
+-- | The package id of the clash-prelude we were built with
+preludePkgId :: String
+preludePkgId = $(lift $ pkgIdFromTypeable $ defSyn "")
+
+-- | Check that we're using the same clash-prelude as we were built with
+--
+-- Because if they differ clash won't be able to recognize any ANNotations.
+checkForInvalidPrelude :: Monad m => HscTypes.ModGuts -> m ()
+checkForInvalidPrelude guts = case filter isWrongPrelude pkgIds of
+                           [] -> return ()
+                           (x:_) -> throw (ClashException noSrcSpan (msgWrongPrelude x) Nothing)
```

Comment:
Did you mean this indention to start there? It seems more or less random. I usually prefer:

```haskell
checkForInvalidPrelude guts = 
  case filter isWrongPrelude pkgIds of
    [] -> return ()
    (x:_) -> throw (ClashException noSrcSpan (msgWrongPrelude x) Nothing)
```

or even

```haskell
checkForInvalidPrelude guts = 
  case filter isWrongPrelude pkgIds of
    [] -> 
      return ()
    (x:_) -> 
      throw (ClashException noSrcSpan (msgWrongPrelude x) Nothing)
```

Reply:
Thnx
At some point is was nicely indented with respect to the case, then refactoring happened and the name of the function got longer.
I'll fix it tomorrow

Reply:
Ah right! That's why I usually prefer to write it as my second suggestion; it really helps to reduce how much (annoying) realignment you need to do when refactoring. Maybe one day it'll be part of The Official Clash Style Guide :grin: